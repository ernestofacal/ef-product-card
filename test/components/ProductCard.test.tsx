import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product1 } from '../data/products';
describe('ProductCard', () => {
  test('1-ProductCard debe de mostrar el componente correctamente con un hijo', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <h1></h1>}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('1-ProductCard debe incrementar el contador', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {({ count, increaseBy }) => (
          <>
            <h1>ProductCard</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(1)}>boton</button>
          </>
        )}
      </ProductCard>
    );
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
    act(() => {
      (tree as any).children[2].props.onClick();
    });
    tree = wrapper.toJSON();
    expect((tree as any).children[1].children[0]).toBe('1');
  });
});
