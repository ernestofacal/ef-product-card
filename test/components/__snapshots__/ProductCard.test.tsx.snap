// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard 1-ProductCard debe de mostrar el componente correctamente con un hijo 1`] = `
<div
  className="productCard undefined"
>
  <h1 />
</div>
`;

exports[`ProductCard 1-ProductCard debe incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCard
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    boton
  </button>
</div>
`;
